decidim-core/app/controllers/concerns/decidim/participatory_space_context.rb

```diff
@@ -56,5 +56,12 @@ def ability_context
     def layout
       current_participatory_space_manifest.context(current_participatory_space_context).layout
     end
+
+    # Method for current user can visit the space (assembly or proces)
+    def current_user_can_visit_space?
+      (current_participatory_space.private_space? &&
```

Comment:
Not all spces might respond to the `private_space?` method. Use `try(:private_space?)`, or add a convenience method inside the `Decidim::Participable` concern so that spaces can override it conveniently.